pkg/image/importer/importer.go

```diff
@@ -372,6 +369,64 @@ func (isi *ImageStreamImporter) calculateImageSize(ctx gocontext.Context, repo d
 	return nil
 }
 
+// Defaults for converting manifest lists to schema1/schema2 manifests.
+// See https://github.com/docker/distribution/blob/06fa77aa11a3913096efcb9b5bd25db8ef55a939/registry/handlers/manifests.go#L25
+const (
+	defaultArch = "amd64"
+	defaultOS   = "linux"
+)
```

Comment:
After some counseling from @smarterclayton, i'm not satisfied with this solution.  While true that it makes things no worse than they were earlier this week (before the manifestlists were introduced), we need to at least make sure we can tolerate manifestlists that do not have a linux/amd64 entry (And by tolerate i mean, allow users to consume those images, not just "not fail").  So that means the default arch/os needs to be user-selectable, even if only via an undocumented annotation(on the imagestreamtag) that we can hand out to users who run into this problem.

True manifestlist support is too far down the road for us to be able to wait for it to solve this problem, we need a more medium-term solution.

Reply:
Just for a record: manifest lists were introduced by Docker in June 2016.

> we need to at least make sure we can tolerate manifestlists that do not have a linux/amd64 entry

Why do we need this? Do we officially support clusters with Windows nodes? I ask this, because we have this problem for more than 1 year.

Reply:
We need it so that we can limp along until we get proper manifestlist support in place.  We've had the problem for more than a year, it finally got us, and now we should fix it in such a way that we don't trip over the next scenario without at least a workaround in the code that we can offer users.

Reply:
I'm cautioning because we're about to go put a change into place that, while it fixes the problem, doesn't prevent the next problem.  We are getting close to actually having the requirement to do all of these things:

1. run linux/power64le clusters (would be broken if we hardcode linux/amd64)
2. support mixed mode clusters (windows and linux)
3. the possibility that anyone in origin running on non-amd64 is broken when we make this change

It's good to get a quick fix in.  We probably also should be defaulting arch and OS to the compiled version (goos and goarch) rather than hardcoding linux/amd64, although we need to think a little bit about how that could break non-amd64 issues.